frontend/src/pages/CourseSelector/courseDescription/CourseDescription.jsx

```diff
@@ -91,17 +86,18 @@ export default function CourseDescription({ structure }) {
       }
     };
 
-    const getPathToCoursesById = async (id) => {
+    const getPathToCoursesById = async (c) => {
       const [data, err] = await axiosRequest(
         "post",
-        `/courses/coursesUnlockedWhenTaken/${id}`,
-        prepareUserPayload(degree, planner)
+        `/courses/coursesUnlockedWhenTaken/${c}`,
+        prepareUserPayload(degree, planner),
       );
-      console.log(data);
-      if (!err) setCoursesPathTo({
-        "direct_unlock": data.direct_unlock,
-        "indirect_unlock": data.indirect_unlock
-      });
+      if (!err) {
+        setCoursesPathTo({
+          direct_unlock: data.direct_unlock,
+          indirect_unlock: data.indirect_unlock,
+        });
+      }
     };
 
     setpageLoaded(false);
```

Comment:
I checked prod and when clicking the `Add to planner` button, everything seems smooth. Where as with this branch, clicking the `Add to planner` button, the course description rerenders and briefly shows the loading skeleton due to the `pageLoaded` state (as it was set to false before setting it to true after finishing calling the endpoint). May not offer the best UX experience as the page loading is dependent on the network speed.

This may have been a bug before you eslinted everything and it wasn't picked up as there were less dependencies in the useEffect call?

Reply:
yea, i think that was it. We can probably figure out a way to make it not refresh if we modularise a bit more.

Reply:
Yea all right. I was thinking since this was kinda out of the scope of the eslint ticket (since its more of a logic issue), we can try to create a bug ticket for it for the FE team to fix it in another branch, since this ticket is already big enough haha. But totally up fine fr you to make the call on that 😄. Otherwise, everything with the eslint changes looks good to me!

Reply:
@gtangelo https://compclub.atlassian.net/browse/CF-186